client/src/components/AudioWaveform/AudioWaveform.jsx

```diff
@@ -25,44 +26,89 @@ class AudioWaveform extends Component {
         let audioUrl = this.props.audioUrl;
         let height = this.props.height;
         let _component = this;
-        await fetch(audioUrl)
-            .then((response) => {
-                return response.blob();
-            })
-            .then((blob) => {
-                return new Response(blob).arrayBuffer()
-            })
-            .then((buffer) => {
-                return this.audioContext.decodeAudioData(buffer);
-            })
-            .then((audioBuffer) => {
-                var options = {
-                    containers: {
+
+        if (audioUrl === 'Loop') {
+            // create some dummy data
+            let toneNumBars = Tone.Time(this.props.room.numBars, 'm');
+            let numSeconds = toneNumBars.toSeconds();
+            let numberOfChannels = 1;
+            let sampleRate = this.audioContext.sampleRate;
+
+            var myArrayBuffer = this.audioContext.createBuffer(numberOfChannels, numSeconds*sampleRate, sampleRate);
+            for (var channel = 0; channel < myArrayBuffer.numberOfChannels; channel++) {
+                // This gives us the actual array that contains the data
+                var nowBuffering = myArrayBuffer.getChannelData(channel);
+                for (var i = 0; i < myArrayBuffer.length; i++) {
+                    nowBuffering[i] = (Math.random() * 2 - 1) * 0.001;
+                }
+            }
+            var options = {
+                containers: {
+                    // zoomview: this.refs.zoomView,
+                    overview: this.refs[`overView-${audioName}`]
+                },
+                height: height,
+                overviewWaveformColor: 'navy',
+                player: new Player(myArrayBuffer),
+                webAudio: {
+                    audioBuffer: myArrayBuffer,
+                    scale: 128,
+                    multiChannel: false
+                },
+                keyboard: true,
+                showPlayheadTime: true,
+                zoomLevels: [128, 256, 512, 1024, 2048, 4096]
+            };
+
```

Comment:
This and peaks.init is same code as in the else block? Can you just pull both outside of the if statement